xbmc/utils/ColorUtils.cpp

```diff
@@ -0,0 +1,29 @@
+/*
+ *      Copyright (C) 2005-2018 Team XBMC
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "Color.h"
+#include "ColorUtils.h"
+#include "math.h"
+
+UTILS::Color ColorUtils::ChangeOpacity(UTILS::Color color, const float opacity)
+{
+  int newAlpha = ceil( ((color >> 24) & 0xff) * opacity);
+  return color += (newAlpha << 24);
```

Comment:
I don't think you want += here, as the assignment is not used after this returns